### PR TITLE
feat: Create release process 

### PR DESCRIPTION
**Pull request by dbadura**
**Description**

Changes proposed in this pull request:

- Create release automation for busola

**Related issue(s)**

**Definition of done**

- [x] The PR's title starts with one of the following prefixes:
  - feat: A new feature
  - fix: A bug fix
  - docs: Documentation only changes
  - refactor: A code change that neither fixes a bug nor adds a feature
  - test: Adding tests
  - chore: Maintainance changes to the build process or auxiliary tools, libraries, workflows, etc.
- [x] Related issues are linked. To link internal trackers, use the issue IDs like `backlog#4567`
- [x] Explain clearly why you created the PR and what changes it introduces
- [x] All necessary steps are delivered, for example, tests, documentation, merging
https://github.com/kyma-project/busola/issues/3375

### CODE REVIEW

File: .github/workflows/busola-backend-build.yml [hunk 1]
```diff
@@ -25,6 +31,4 @@ jobs:
     with:
       name: busola-backend
       dockerfile: Dockerfile
-      context: backend
-      export-tags: true
-      tags: latest
```
Comment (mrCherry97) [inline .github/workflows/busola-backend-build.yml:30]: make sure that we can remove this

File: .github/workflows/busola-web-build.yml [hunk 2]
```diff
@@ -36,6 +41,5 @@ jobs:
     with:
       name: busola-web
       dockerfile: Dockerfile.web
-      context: .
-      export-tags: true
-      tags: latest
+      tags: ${{ inputs.tag != '' && inputs.tag || 'latest' }}
+      build-args: ${{ inputs.tag != '' && format('tag={0}', inputs.tag) || '' }}
```
Comment (mrCherry97) [inline .github/workflows/busola-web-build.yml:45]: Make sure after first release on dev, that it is working and displayed correctly

File: .github/workflows/create-release.yml [hunk 3]
```diff
@@ -17,143 +17,67 @@ permissions: # used by build images steps
...
-  #
-  #          git add .
-  #          git commit --allow-empty -m "upgrade dependencies"
-  #          git push origin ${{ github.ref_name }}
+  #  integrations:
+  #    needs: create-draft
+  #    secrets: inherit
+  #    uses: ./.github/workflows/_integration-tests.yaml
+  #    with:
+  #      image: europe-docker.pkg.dev/kyma-project/prod/serverless-operator:${{ github.event.inputs.name }}
```
Comment (mrCherry97) [inline .github/workflows/create-release.yml:36]: remove it for now, will be done in separate task

File: .github/workflows/create-release.yml [hunk 4]
```diff
@@ -17,143 +17,67 @@ permissions: # used by build images steps
...
 
-  create-tag:
-    name: Create tag
-    #    needs: upgrade-images
+  create-release:
+    name: Create release
+    needs: [ build-web, build-backend ]
```
Comment (mrCherry97) [inline .github/workflows/create-release.yml:40]:
```suggestion
    needs: [ build-web, build-backend, build-local ]
```

File: Dockerfile.web [hunk 5]
```diff
@@ -15,7 +16,7 @@ ENV CI true
 
 COPY . /app
 
-RUN yq -i '.version = "'${default_tag}'"' public/version.yaml && \
+RUN export TAG=${tag:-$default_tag} &&  yq -i '.version = "'${TAG}'"' public/version.yaml && \
```
Comment (mrCherry97) [inline Dockerfile.web:19]: Add some comment to clarify what's going on

File: resources/apply-resources.sh [hunk 6]
```diff
@@ -15,7 +15,8 @@ TMP_DIR="../temp/resources"
 mkdir -p "${TMP_DIR}"
 cp -rf . "${TMP_DIR}"
 
-envsubst < "${TMP_DIR}"/base/ingress.tpl.yaml > "${TMP_DIR}"/base/ingress.yaml
-
 kubectl create namespace "${NAMESPACE}" || true
 kubectl apply -k "${TMP_DIR}/environments/${ENVIRONMENT}" --namespace="${NAMESPACE}"
+
+envsubst < "${TMP_DIR}"/ingress/ingress.tpl.yaml > "${TMP_DIR}"/ingress/ingress.yaml
+kubectl apply -k "${TMP_DIR}"/ingress --namespace=$NAMESPACE
```
Comment (mrCherry97) [inline resources/apply-resources.sh:22]: Will be changed with https://github.com/kyma-project/busola/issues/3493

**Review by mrCherry97 — CHANGES_REQUESTED** [review]
Check after merge if `latest` is build, and user can run Busola via docker using latest images.

Check internal docs `deploy-PR-on-cluster` if changes to kustomization will break our previous flow.

File: .github/workflows/create-release.yml [hunk 7]
```diff
@@ -17,143 +17,67 @@ permissions: # used by build images steps
...
     runs-on: ubuntu-latest
-
     steps:
       - name: Checkout code
         uses: actions/checkout@v4
         with:
           fetch-depth: 0
-          ref: ${{ github.ref_name }} # Checkout to latest branch changes
+          ref: ${{ github.ref_name }} # checkout to latest branch changes ( by default this action checkouts to the SHA that triggers action )
```
[the remark this answers is not in the document]
Comment (dbadura) [reply .github/workflows/create-release.yml:47]: Yes, according to docs:
```
# Number of commits to fetch. 0 indicates all history for all branches and tags.
    # Default: 1
    fetch-depth: ''
    ```
If we want to create changelog, it's needed

File: .github/workflows/busola-backend-build.yml (hunk 1 above)
Comment (dbadura) [reply .github/workflows/busola-backend-build.yml]: `export-tags` is not needed in my opinion. According to docs:
```
description: Export parsed tags as build-args into dockerfile. Each tag will have format TAG_x, where x is the tag name passed along with the tag
```
I think that it's unnecessary complication and it hides the build logic.

**Review by mrCherry97 — COMMENTED** [review]
LGTM, waiting for second review to approve

**Review by mrCherry97 — CHANGES_REQUESTED** [review]
Please add documentation: https://github.com/kyma-project/busola/issues/3375#issuecomment-2499949784

**Comment by dbadura** [comment]
> Please add documentation: [#3375 (comment)](https://github.com/kyma-project/busola/issues/3375#issuecomment-2499949784)

I will create separate PR with docs and needed adjustment to busola and configs

**Review by mrCherry97 — APPROVED** [review]

**Pull request merged**